packages/testcontainers/src/utils/port.test.ts

```diff
@@ -0,0 +1,75 @@
+import { describe, expect } from "vitest";
+import { getContainerPort, getProtocol, hasHostBinding } from "./port";
+
+describe("port utilities", () => {
```

Comment:
There are compilation issues in this test suite after the API change, e.g. `parses port from string`.

Another one:

```ts
it("throws error for invalid port format", () => {
  expect(() => getContainerPort("invalid")).toThrow("Invalid port format: invalid");
});
```

Actually it's no longer possible to have this runtime error as we now catch it at compile time

Reply:
Removed all broken tests. Sorry for not catching this before pushing.